packages/block-editor/src/store/reducer.js

```diff
@@ -718,7 +718,20 @@ const BLOCK_SELECTION_INITIAL_STATE = {
 export function blockSelection( state = BLOCK_SELECTION_INITIAL_STATE, action ) {
 	switch ( action.type ) {
 		case 'CLEAR_SELECTED_BLOCK':
-			return BLOCK_SELECTION_INITIAL_STATE;
+			return {
+				...BLOCK_SELECTION_INITIAL_STATE,
+				previousSelection: {
```

Comment:
How about using a simpler syntax like with `omit` from lodash:
```js
return {
				...BLOCK_SELECTION_INITIAL_STATE,
				previousSelection: omit( state, [ 'previousSelection' ] ),
};
```
It would simplify maintanance if the state shape changes in the future.

Can we also add a simple unit test which covers this action in the reducer?

Reply:
Some questions:

 - When do we clear this cache? Do we keep the previous selection indefinitely?
 - What about the other actions that change the selection (SELECT_BLOCK, INSERT_BLOCK...), do we want to save the previous selection there as well?

Reply:
Yes, there might be an edge case based on the failing e2e test. I think when you have a selection and switch to the Code editor and then back to the Visual editor, it might try to restore the selection. It needs to be further investigated as it might break when you remove all blocks in the Code editor mode.

Reply:
@youknowriad yes we can use restoreSelectedBlock in many places but I suggest not in this PR. One example, undo and redo could also take selection into account and return focus: I have a focus in a Paragraph block and I use the inserter to add an Image, when I undo it would be nice to have the focus returned (or selection reapplied).

But I think they should be incremental code changes that store the previous selection where it is needed and then use restoreSelectedBlock. This PR is only about the inspector tho.

Indeed, I will investigate places where the previousSelection cache has to be cleared so it will not result in errors.